apm-agent-api/src/main/java/co/elastic/apm/api/ElasticApm.java

```diff
@@ -83,6 +84,60 @@ private static Object doStartTransaction() {
         return null;
     }
 
+    /**
+     * Similar to {@link ElasticApm#startTransaction()} but creates this transaction as the child of a remote parent.
+     *
+     * <p>
+     * Example:
+     * </p>
+     * <pre>
+     * Transaction transaction = ElasticApm.startTransactionWithRemoteParent(request::getHeader);
+     * </pre>
+     * <p>
+     * Note: If the protocol supports multi-value headers, use {@link #startTransactionWithRemoteParent(HeaderExtractor, HeadersExtractor)}
+     * </p>
+     *
+     * @param headerExtractor a function which receives a header name and returns the fist header with that name
+     * @return the started transaction
+     * @since 1.3.0
+     */
+    @Nonnull
+    public static Transaction startTransactionWithRemoteParent(final HeaderExtractor headerExtractor) {
+        return startTransactionWithRemoteParent(headerExtractor, null);
+    }
+
+    /**
+     * Similar to {@link ElasticApm#startTransaction()} but creates this transaction as the child of a remote parent.
+     *
+     * <p>
+     * Example:
+     * </p>
+     * <pre>
+     * Transaction transaction = ElasticApm.startTransactionWithRemoteParent(request::getHeader, request::getHeaders);
+     * </pre>
+     * <p>
+     * Note: If the protocol does not support multi-value headers, use {@link #startTransactionWithRemoteParent(HeaderExtractor)}
+     * </p>
+     * <p>
+     *
+     * @param headerExtractor a function which receives a header name and returns the fist header with that name
+     * @param headersExtractor  a function which receives a header name and returns all headers with that name
+     * @return the started transaction
+     * @since 1.3.0
+     */
+    @Nonnull
+    public static Transaction startTransactionWithRemoteParent(HeaderExtractor headerExtractor, HeadersExtractor headersExtractor) {
```

Comment:
even for protocols which do support multiple headers, we want to have the ability to get a single value for a header in case we know it can only have a single value, like `traceparent`.